base/reflection.jl

```diff
@@ -281,13 +289,19 @@ end
 
 function return_types(f::Function, types::ANY=Tuple)
     types = to_tuple_type(types)
-    rt = []
+    rt = Any[]
     for x in _methods(f,types,-1)
+        # if the method is a @generated function and the argument types are not concrete,
+        # skip the method as the generated function cannot be instantiated
```

Comment:
technically, the return type of that method is `Any`, so it might be worth noting that in the result?